controllers/cluster_controller.go

```diff
@@ -143,6 +143,15 @@ func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return ctrl.Result{}, nil
 	}
 
+	if cluster.Spec.BundlesRef == nil {
+		if err := r.SetBundlesRef(ctx, cluster); err != nil {
+			if apierrors.IsNotFound(err) {
+				return reconcile.Result{}, nil
```

Comment:
Are you doing this because other parts of the controller do the same? If that's the case, either those other parts are wrong or they have a valid reason to do it. I don't think there is one in this case, but I might be missing something :)

I think we should return an error for this scenario as well